src/util/union_find.cpp

```diff
@@ -80,13 +80,13 @@ void unsigned_union_find::re_root(size_type old_root, size_type new_root)
   old_root=find(old_root);
 
   // same set?
-  // assert(find(new_root)==old_root);
+  // assert(find(new_root)==old_root); // Commented out dead code? To review
```

Comment:
Yes, should be removed - note that the code below excludes this case without an error.

At least I think so, I'd be happy for someone more knowledgeable about this particular implementation to chime in on this.